autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/cleanup/ActivityCleanupEvaluator.java

```diff
@@ -0,0 +1,58 @@
+package com.sequenceiq.periscope.monitor.evaluator.cleanup;
+
+import java.util.Objects;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.periscope.model.ScalingActivities;
+import com.sequenceiq.periscope.monitor.context.EvaluatorContext;
+import com.sequenceiq.periscope.monitor.context.ScalingActivitiesEvaluatorContext;
+import com.sequenceiq.periscope.monitor.evaluator.EvaluatorExecutor;
+import com.sequenceiq.periscope.monitor.evaluator.EventPublisher;
+import com.sequenceiq.periscope.monitor.event.ScalingActivityCleanupEvent;
+
+@Component("ActivityCleanupEvaluator")
+@Scope("prototype")
+public class ActivityCleanupEvaluator extends EvaluatorExecutor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ActivityCleanupEvaluator.class);
+
+    private static final String EVALUATOR_NAME = ActivityCleanupEvaluator.class.getName();
+
+    @Inject
+    private EventPublisher eventPublisher;
+
+    private ScalingActivities scalingActivities;
+
+    @Nonnull
+    @Override
+    public EvaluatorContext getContext() {
+        return new ScalingActivitiesEvaluatorContext(scalingActivities);
+    }
+
+    @Override
+    public String getName() {
+        return EVALUATOR_NAME;
+    }
+
+    @Override
+    public void setContext(EvaluatorContext context) {
+        this.scalingActivities = (ScalingActivities) context.getData();
+    }
+
+    @Override
+    protected void execute() {
+        if (Objects.isNull(scalingActivities)) {
+            LOGGER.info("No scalingActivities found");
+            return;
+        }
+        LOGGER.info("Publishing event within ActivityCleanupEvaluator for {} scalingActivities", scalingActivities.getActivityIds().size());
+        eventPublisher.publishEvent(new ScalingActivityCleanupEvent(scalingActivities));
```

Comment:
Why publish a separate event for this, instead of performing cleanup here itself ?

Reply:
I followed the pattern in the existing code to cast an event and use corresponding event listeners to make DB transactions. Changed this to directly update the DB from the evaluator